pkg/apis/eventing/v1alpha1/channel_types.go

```diff
@@ -139,11 +143,28 @@ func (cs *ChannelStatus) InitializeConditions() {
 // MarkProvisioned sets ChannelConditionProvisioned condition to True state.
 func (cs *ChannelStatus) MarkProvisioned() {
 	chanCondSet.Manage(cs).MarkTrue(ChannelConditionProvisioned)
+
+	// Channel-default-controller sets ChannelConditionProvisionerInstalled=False, and it needs to be set to True by individual controllers
+	// This is done so that each individual channel controller gets it for free.
+	cs.MarkProvisionerInstalled()
 }
 
 // MarkNotProvisioned sets ChannelConditionProvisioned condition to False state.
 func (cs *ChannelStatus) MarkNotProvisioned(reason, messageFormat string, messageA ...interface{}) {
 	chanCondSet.Manage(cs).MarkFalse(ChannelConditionProvisioned, reason, messageFormat, messageA...)
+	// Channel-default-controller sets ChannelConditionProvisionerInstalled=False, and it needs to be set to True by individual controllers
+	// This is done so that each individual channel controller gets it for free.
+	cs.MarkProvisionerInstalled()
+}
+
+// MarkProvisionerInstalled sets ChannelConditionProvisionerInstalled condition to True state.
+func (cs *ChannelStatus) MarkProvisionerInstalled() {
+	chanCondSet.Manage(cs).MarkTrue(ChannelConditionProvisionerInstalled)
```

Comment:
I don't think you want to call `MarkTrue`. This will also set the `Ready` condition. And the goal of the `ProvisionerInstalled` is to give feedback to the user that there is no controller operating on this resource. 

I think you should use [SetCondition](https://github.com/knative/pkg/blob/267197692a3e9d011f15aec20ad0b587d9650f06/apis/duck/v1alpha1/condition_set.go#L62) to avoid the side-effect of setting `Ready`

Reply:
If you do this change the method should be renamed to `SetProvisionerInstalled`

Reply:
I don't think we should do that. Ready or Happy condition will be called only if all its dependent conditions are True. In this case Ready is True if each one of ChannelConditionProvisioned, ChannelConditionAddressable, ChannelConditionProvisionerInstalled is True.
And this is what the intent it. Just setting ProvisionerInstalled=True will not set Ready=True unless all other conditions are true.